File: tools/check_abi.c
#include <stdio.h>
#include <stddef.h>
#include "../bindings/c/qwd.h"

int main() {
    printf("--- C TELEMETRY ABI REPORT ---\n");
    printf("Total Size: %zu bytes\n", sizeof(qwd_telemetry_t));
    
    printf("\nFIELD OFFSETS:\n");
    printf("read_count:   %zu\n", offsetof(qwd_telemetry_t, read_count));
    printf("total_bases:  %zu\n", offsetof(qwd_telemetry_t, total_bases));
    printf("status:       %zu\n", offsetof(qwd_telemetry_t, status));
    printf("gc_dist:      %zu\n", offsetof(qwd_telemetry_t, gc_distribution));
    printf("len_dist:     %zu\n", offsetof(qwd_telemetry_t, length_distribution));
    printf("heatmap:      %zu\n", offsetof(qwd_telemetry_t, quality_heatmap));
    
    return 0;
}

File: tools/check_abi.zig
const std = @import("std");
const qwd = @import("qwd_api");

pub fn main() void {
    const t = qwd.qwd_telemetry_t;
    std.debug.print("--- ZIG TELEMETRY ABI REPORT ---\n", .{});
    std.debug.print("Total Size: {d} bytes\n", .{@sizeOf(t)});
    std.debug.print("Alignment: {d} bytes\n", .{@alignOf(t)});
    std.debug.print("\nFIELD OFFSETS:\n", .{});
    std.debug.print("read_count:   {d}\n", .{@offsetOf(t, "read_count")});
    std.debug.print("total_bases:  {d}\n", .{@offsetOf(t, "total_bases")});
    std.debug.print("status:       {d}\n", .{@offsetOf(t, "status")});
    std.debug.print("gc_dist:      {d}\n", .{@offsetOf(t, "gc_distribution")});
    std.debug.print("len_dist:     {d}\n", .{@offsetOf(t, "length_distribution")});
    std.debug.print("heatmap:      {d}\n", .{@offsetOf(t, "quality_heatmap")});
}
